File: 118-pascals-triangle/pascals-triangle.ts
function generate(numRows: number): number[][] {
    if(numRows===0)
        return []
    if(numRows===1)
        return [[1]]
   
    let pascalTriangle:number[][] = [
        [1],[1,1]
       
    ]
    for(let i=2;i<numRows;i++){
        let tab:number[] =[]
        tab[0] = 1
        tab[i] = 1
       for(let j=1;j<i;j++){
         tab[j] = pascalTriangle[i-1][j-1]+pascalTriangle[i-1][j]
       }
       pascalTriangle.push(tab)
    }
    return pascalTriangle
};

File: 118-pascals-triangle/pascals-triangle.py


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        if numRows == 1:
            return [[1]]
        
        pascalTriangle = [[1], [1, 1]]
        
        for i in range(2, numRows):
            tab = [0] * (i + 1)
            tab[0] = 1
            tab[i] = 1
            for j in range(1, i):
                tab[j] = pascalTriangle[i - 1][j - 1] + pascalTriangle[i - 1][j]
            pascalTriangle.append(tab)
        
        return pascalTriangle
